RAM-27

brunoliratm opened the pull request:
This pull request introduces several changes to the Rick and Morty API project, focusing on enhancing the `EpisodeController` and related classes to support pagination and improve response handling. The most important changes include updating the `EpisodeController` to return `ResponseEntity`, modifying the `EpisodeService` to handle paginated responses, and refining the `EpisodeDto` to use `JsonProperty` annotations.

### Enhancements to EpisodeController:

* [`src/main/java/com/rickmorty/Controllers/EpisodeController.java`](diffhunk://#diff-28e03c434a7550cd0f59e1677ea7899791c7ef1bf72c6ae87a81642f6cf0ef35L21-L33): Updated `getAllEpisodes` and `getEpisodeById` methods to return `ResponseEntity` instead of raw objects. This improves the response structure and status handling.

### Pagination and Response Handling:

* [`src/main/java/com/rickmorty/Services/EpisodeService.java`](diffhunk://#diff-76373ce3d869a63290178d1f081bb3d5ef5b8456152f07e230d0e2e0999e6362R7-R101): Added pagination support to `findAllEpisodes` method and improved error logging. Introduced helper methods to rewrite API responses and episode data.

### DTO Improvements:

* [`src/main/java/com/rickmorty/DTO/EpisodeDto.java`](diffhunk://#diff-f91c7f403100b77f1cf30bb1fe1ad2b76f46c12c8628c59b3281d4d00a8700fbL5-R15): Simplified the `EpisodeDto` class by using `JsonProperty` annotations and removing unnecessary constructor logic.

### Minor Cleanups:

* [`src/main/java/com/rickmorty/Controllers/CharacterController.java`](diffhunk://#diff-be7e49e511c94fa1dbcf2ed7aca83ac8d132c5b67a116e4c38454417e2690226L6): Removed an unnecessary blank line.
* [`src/main/java/com/rickmorty/Models/EpisodeModel.java`](diffhunk://#diff-845ffa953516ed9c7cfc19bdfe446cd6900cbb3032b67fc85397f7fab9a29f89L4-L26): Removed unused imports and fields.

### Documentation Update:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R73): Added documentation for the new paginated episodes endpoint.

Paulo-Araujo-Jr reviewed (approved):
Excelente trabalho nas alterações feitas! A implementação está clara e bem estruturada. Tudo está pronto para ser integrado ao projeto. Pull request aprovado! 🚀

MrMesquita commented on src/main/java/com/rickmorty/Services/EpisodeService.java:
```diff
@@ -4,53 +4,100 @@
+        return new ApiResponseDto<>(updatedInfo, updatedResults);
+    }
+
+    private InfoDto rewriteInfoDto(InfoDto originalInfo, Integer page) {
```
Remover parâmetro não usado Interger page, aproveita e tira de characterService tbm

MrMesquita commented on src/main/java/com/rickmorty/Services/EpisodeService.java:
```diff
@@ -4,53 +4,100 @@
+        InfoDto updatedInfo = rewriteInfoDto(apiResponseDto.info(), null);
+
+        List<EpisodeDto> updatedResults = new ArrayList<>();
+        for (EpisodeDto episode : apiResponseDto.results()) {
```
utiliza o map aqui como fizesse em rewriteEpisodeDto

brunoliratm commented:
### Documentation Updates:
* Enhanced the `README.md` with badges, a table of contents, and an updated project description.
* Translated sections from Portuguese to English and added a roadmap section. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L32-R78) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L68-R131)

### Service Enhancements:
* Updated `EpisodeService` to support pagination and refactored methods for better readability and error handling.
* Improved `CharacterService` to use `Optional` for URL rewriting and enhanced the `rewriteApiResponse` method. [[1]](diffhunk://#diff-7709e8d9309662b339743abdc8fb83c093a95d8c7e594c8765aa67aab1f8b1e5L91-R92) [[2]](diffhunk://#diff-7709e8d9309662b339743abdc8fb83c093a95d8c7e594c8765aa67aab1f8b1e5L101-R111)
